### PR TITLE
Phase3 connect resturants and cuisine

### PR DESCRIPTION
**Pull request by yardencohen**
screen shot 2017-03-30 at 15 30 50

backend and basic front-end

### CODE REVIEW

File: app/controllers/cuisines_controller.rb [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
+    def set_cuisine
+      @cuisine = Cuisine.find(params[:id])
+    end
+
+    # Never trust parameters from the scary internet, only allow the white list through.
+    def cuisine_params
+      params.require(:cuisine).permit(:title, :image)
```
Comment (RonaTross) [inline app/controllers/cuisines_controller.rb:72]: awesome

File: app/controllers/cuisines_controller.rb [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  def destroy
+    @cuisine.destroy
+    respond_to do |format|
+      format.html { redirect_to cuisines_url, notice: 'Cuisine was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
```
Comment (RonaTross) [inline app/controllers/cuisines_controller.rb:65]: no need to indent all the methods after `private`

File: app/controllers/resturants_controller.rb [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  def edit
+  end
+
+  # POST /resturants
+  # POST /resturants.json
+  def create
+    @resturant = Resturant.new(resturant_params)
+
+    respond_to do |format|
+      if @resturant.save
```
Comment (RonaTross) [inline app/controllers/resturants_controller.rb:30]: 👍

File: app/controllers/resturants_controller.rb [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  def destroy
+    @resturant.destroy
+    respond_to do |format|
+      format.html { redirect_to resturants_url, notice: 'Resturant was successfully destroyed.' }
+      format.json { head :no_content }
+    end
+  end
+
+  private
+    # Use callbacks to share common setup or constraints between actions.
```
Comment (RonaTross) [inline app/controllers/resturants_controller.rb:65]: same

File: app/models/resturant.rb [hunk 5]
```diff
@@ -0,0 +1,8 @@
+class Resturant < ApplicationRecord	
+	belongs_to :cuisine
+	validates :name, presence: true
+	validates :cuisine_id, presence: true
+	validates :rating, presence: true
```
Comment (RonaTross) [inline app/models/resturant.rb:5]: can we also validate rating is in the correct range? also take a look at `enum`

File: db/migrate/20170327104900_create_resturants.rb [hunk 6]
```diff
@@ -0,0 +1,14 @@
+class CreateResturants < ActiveRecord::Migration[5.0]
+  def change
+    create_table :resturants do |t|
```
Comment (RonaTross) [inline db/migrate/20170327104900_create_resturants.rb:3]: I'd add database level validations on top of the model-level ones (like can't be null, needs to be unique etc.)

File: app/models/resturant.rb (hunk 5 above)
Comment (yardencohen) [reply app/models/resturant.rb:5]: I later removed this from restaurants and added it to reviews where I also added this validation

File: db/migrate/20170327105842_change_max_delivery_time.rb [hunk 7]
```diff
@@ -0,0 +1,10 @@
+class ChangeMaxDeliveryTime < ActiveRecord::Migration[5.0]
+  def change
+  	reversible do |dir|
```
Comment (odedmagger) [inline db/migrate/20170327105842_change_max_delivery_time.rb:3]: Cool! I think it's more common here to implement `up` and `down`, it's no big deal that you repeat `change_table :restaurants`

Did you try to migrate and rollback, and see if this works?

File: db/migrate/20170330091029_add_cuisine_to_resturants.rb [hunk 8]
```diff
@@ -0,0 +1,5 @@
+class AddCuisineToResturants < ActiveRecord::Migration[5.0]
```
Comment (odedmagger) [inline db/migrate/20170330091029_add_cuisine_to_resturants.rb:1]: In a real life scenario, we would unify adding cuisine and removing genre in one atomic migration.

We would also write a code that takes all of the existing genres and creating cuisines from them, before deleting the genre field.

No need to do it in the training project as we don't care about the persistence of the data :)

File: test/controllers/resturants_controller_test.rb [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  end
+
+  test "should get new" do
+    get new_resturant_url
+    assert_response :success
+  end
+
+  test "should create resturant" do
+    assert_difference('Resturant.count') do
+      post resturants_url, params: { resturant: { tenbis: @resturant.tenbis, address: @resturant.address, genre: @resturant.genre, max_delivery_time: @resturant.max_delivery_time, name: @resturant.name, rating: @resturant.rating } }
```
Comment (odedmagger) [inline test/controllers/resturants_controller_test.rb:20]: Does this still work, now that you changed genre to cuisine?

File: app/controllers/cuisines_controller.rb [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+
+  # POST /cuisines
+  # POST /cuisines.json
+  def create
+    @cuisine = Cuisine.new(cuisine_params)
+
+    respond_to do |format|
+      if @cuisine.save
+        format.html { redirect_to @cuisine, notice: 'Cuisine was successfully created.' }
+        format.json { render :show, status: :created, location: @cuisine }
```
Comment (odedmagger) [inline app/controllers/cuisines_controller.rb:32]: Why is the parameter called `location`?

**Review by odedmagger — APPROVED** [review]
Looking good, though I suspect you may have broken the create call when you changed genre to cuisine. Please check that you can still create new restaurants and get the expected cuisine.

**Pull request merged**